node/external/transactionAPI/unmarshalAndPrepare.go

```diff
@@ -0,0 +1,129 @@
+package transactionAPI
+
+import (
+	"encoding/hex"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	rewardTxData "github.com/ElrondNetwork/elrond-go-core/data/rewardTx"
+	"github.com/ElrondNetwork/elrond-go-core/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+)
+
+type unmarshalerAndPreparer struct {
+	addressPubKeyConverter core.PubkeyConverter
+	marshalizer            marshal.Marshalizer
+}
+
+func newTransactionUnmashalerAndPreparer(marshalizer marshal.Marshalizer, addressPubKeyConverter core.PubkeyConverter) *unmarshalerAndPreparer {
+	return &unmarshalerAndPreparer{
+		marshalizer:            marshalizer,
+		addressPubKeyConverter: addressPubKeyConverter,
+	}
+}
+
+func (up *unmarshalerAndPreparer) unmarshalTransaction(txBytes []byte, txType transaction.TxType) (*transaction.ApiTransactionResult, error) {
+	switch txType {
```

Comment:
can be avoided by using polymorphism
This can be split in at least 4 other small processors and use a common interface

Reply:
Will make a task for this